java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1120,6 +1122,24 @@ private boolean hasLocallyPresentOptionalBinding(ResolvedBindings resolvedBindin
               && !getLocalExplicitBindings(keyFactory.unwrapOptional(resolvedBindings.key()).get())
                   .isEmpty();
         }
+
+        /**
+         * Returns {@code true} if {@code resolvedBindings} contains a provision binding
+         * for which there is an explicit present binding in this component that was resolved
+         * previously.
+         */
+        boolean hasLocallyConflictingContributionBinding(ResolvedBindings resolvedBindings) {
```

Comment:
Actually, we can probably get away without checking the binding kind at all since multibindings are already accounted for and any other kind of duplicate binding would be a conflict.